packages/zoe/src/z2spec/escrow-exo.js

```diff
@@ -0,0 +1,311 @@
+import { isOfferSafe } from '../contractFacet/offerSafety.js';
+import { prepareOwnable } from '../contractSupport/prepare-ownable.js';
+
+/**
+ * @import {Zone} from '@agoric/base-zone'
+ */
+
+const { keys, entries, fromEntries } = Object;
+const { isEmpty } = AmountMath;
+
+const objectFilter = (obj, pred) =>
+  fromEntries(entries(obj).filter(([k, v]) => pred(v, k)));
+
+/**
+ * @param {Zone} zone
+ * @param {<T>() => PublishKit<T>} makeDurablePublishKit
+ */
+export const prepareSeatKit = (zone, makeDurablePublishKit) =>
+  zone.exoClassKit(
+    'Seat Kit',
+    undefined,
+    proposal => {
+      /** @type {PublishKit<unknown>} */
+      const exitKit = makeDurablePublishKit();
+      /** @type {PublishKit<PaymentPKeywordRecord>} */
+      const payoutsKit = makeDurablePublishKit();
+      return {
+        proposal,
+        exitKit,
+        payoutsKit,
+        exiting: false,
+        /** @type {Allocation} */
+        currentAllocation: proposal.give,
+      };
+    },
+    {
+      readOnly: {
+        hasExited() {
+          return this.state.exiting;
+        },
+        getProposal() {
+          return this.state.proposal;
+        },
+        getCurrentAllocation() {
+          return this.state.currentAllocation;
+        },
+        getExitSubscriber() {
+          return this.state.exitKit.subscriber;
+        },
+      },
+
+      /** The right to collect payouts */
+      seat: {
+        /** @param {Keyword} kw */
+        async getPayout(kw) {
+          const { payoutsKit } = this.state;
+          const paid = await payoutsKit.subscriber.getUpdateSince();
+          return paid.value[kw];
+        },
+        async getPayouts() {
+          const { payoutsKit } = this.state;
+          const paid = await payoutsKit.subscriber.getUpdateSince();
+          return paid.value;
+        },
+
+        /** @param {unknown} completion */
+        async exit(completion) {
+          assert.fail('TODO');
+        },
+        fail(reason) {
+          assert.fail('TODO');
+        },
+
+        readOnly() {
+          const { readOnly } = this.facets;
+          return readOnly;
+        },
+
+        getInvitationCustomDetails() {
+          const { exiting, proposal, currentAllocation } = this.state;
+          !exiting || Fail`already exited`;
+
+          return harden({
+            proposal,
+            currentAllocation,
+          });
+        },
+
+        // repeat accessors for compat
+        hasExited() {
+          return this.state.exiting;
+        },
+        getProposal() {
+          return this.state.proposal;
+        },
+        getCurrentAllocation() {
+          return this.state.currentAllocation;
+        },
+        getExitSubscriber() {
+          return this.state.exitKit.subscriber;
+        },
+      },
+      /** The right to allocate payouts (subject to offer safety) */
+      resolver: {
+        readOnly() {
+          const { readOnly } = this.facets;
+          return readOnly;
+        },
```

Comment:
Raises the same privacy expectation issue wrt previous owners finding out about exits after ownership transfer. But in this case, wrt resolver ownership transfer, which I don't think makes the issue either better or worse.